src/controller/controller.ts

```diff
@@ -265,7 +265,9 @@ class Controller extends events.EventEmitter {
 
     public async stop(): Promise<void> {
         this.stopping = true;
-        this.databaseSave();
+        try {
```

Comment:
Is this one really needed? (I don't see how this can fail unless e.g. wrong permission are setup). 

As an alternative we can move it below the adapter.stop().

Reply:
This is **the** key item of this PR.
If the start of the adapter fails before the network is initialised, but after the serial port was opened, some internal structures are not initialised leading to a number of errors. The exact error depends on how far the initialisation has come and possibly also on timing. I was not able to pinpoint the exact placement, which is why i went for the 'catch all' approach.

```
TypeError: Cannot read property 'getEntries' of null
    at Function.loadFromDatabaseIfNecessary (/opt/iobroker/node_modules/zigbee-herdsman/dist/controller/model/device.js:220:55)
```

Moving it behind the adapter.stop() might be possible, but would only solve half the issue. The Herdsman would still fail due to its internal structures not being initialised in a situation where this should not pose an issue to begin with.

Reply:
Can you try without this try/catch and changing https://github.com/Koenkk/zigbee-herdsman/blob/941613e8f75047f529a8f0c87d57192508c9b78d/src/controller/controller.ts#L114 to 


```js
        // Database
        this.database = Database.open(this.options.databasePath);
        Entity.injectDatabase(this.database);

        // Adapter
        this.adapter = await Adapter.create(this.options.network,
            this.options.serialPort, this.options.backupPath, this.options.adapter, this.logger);
        debug.log(`Starting with options '${JSON.stringify(this.options)}'`);
        const startResult = await this.adapter.start();
        debug.log(`Started with result '${startResult}'`);
        Entity.injectAdapter(this.adapter);
```

Reply:
I'll check this later today.

Reply:
tested - works.